pints/_mcmc/__init__.py

```diff
@@ -301,24 +303,54 @@ def __init__(
             self, log_pdf, chains, x0, sigma0=None, transformation=None,
             method=None):
 
-        # Check function
-        if not isinstance(log_pdf, pints.LogPDF):
-            raise ValueError('Given function must extend pints.LogPDF')
+        if isinstance(log_pdf, pints.LogPDF):
+            self._multi_logpdf = False
+
+        else:
+            self._multi_logpdf = True
+            try:
+                log_pdfs_iter = iter(log_pdf)  # noqa
```

Comment:
It's very neat to check for both the iter and len parts of the sequence interface, but perhaps just len() is enough? (Then you can combine both checks: try: if len() != raise; except TypeError)